glib/FastActivator.cs

```diff
@@ -40,6 +40,26 @@ static FastCreateObject FastCtor (Type t, Dictionary<Type, FastCreateObject> cac
 			return method;
 		}
 
+		delegate object FastCreateBoxed (IntPtr ptr);
+		static FastCreateBoxed FastBoxed (Type t, Dictionary<Type, FastCreateBoxed> cache)
+		{
+			FastCreateBoxed method;
+			lock (cache) {
+				if (!cache.TryGetValue (t, out method)) {
+					var newMethod = t.GetMethod ("New", BindingFlags.Static | BindingFlags.Public | BindingFlags.FlattenHierarchy);
+					if (newMethod != null) {
+						var param = Expression.Parameter (typeof (IntPtr));
+						var call = Expression.Call (newMethod, param);
+						var callWithConvert = Expression.Convert (call, typeof (object));
+						cache [t] = method = (FastCreateBoxed)Expression.Lambda (typeof(FastCreateBoxed), callWithConvert, param).Compile ();
```

Comment:
Isn't there an `Expression.Lambda <FastCreateBoxed> (callWithConvert, param).Compile ()` overload we could use?